cmd/packaging/darwin-dmg.go

```diff
@@ -22,6 +22,16 @@ var DarwinDmgTask = &packagingTask{
 		if err != nil {
 			return "", err
 		}
+		appFileOriginalPath := fmt.Sprintf("dmgdir/%s %s.app", applicationName, version)
+		appFileFinalPath := fmt.Sprintf("dmgdir/%s.app", applicationName)
+		cmdRenameApp := exec.Command("mv", appFileOriginalPath, appFileFinalPath)
```

Comment:
You can just move the directory using go

Reply:
using os.Rename now in the updated PR

Reply:
Have also updated the PR with renaming of bundle for darwin-pks (Would need you to confirm if the patch is OK).

You are correct, the specs do mention that the extension should be omitted but when I tested the linux-deb packaging,
without the extension the icon showed up as blank, atleast for me on elementary OS based on ubuntu 18.04,
and adding the extension fixed the issue.

Reply:
I looked at the freedesktop specifications which states that `.desktop` files should have the file extension in the icon path, but appimage states that it shouldn't contain it. So everything is alright